### DISCUSSION TITLE
i can't find out why this simple animation just don't work properly

### DISCUSSION THREAD

**Discussion by bessim-dev** [pmndrs/react-spring#1940] [Support]
i'm trying to build a stagger effect with an array of data and useTransition but it breaks whenever I change the data of the array,
[https://codesandbox.io/s/stagger-qmtc9t?file=/src/App.js](url)

**Comment by joshuaellis** [comment]
Are you able to explain what you mean by "it breaks whenever i change the data of the array", what are you expected etc.

**Reply by bessim-dev** [reply]
hey, i expect it to run the leave animation for the first data and then animate in the second data.

**Reply by bessim-dev** [reply]
1- on enter, the items in the array will animate in (in stagger effect)
2- when I change the index the old items animate out and the new ones animate in

**Reply by joshuaellis** [reply]
looks like there's a conflict between the `trail` prop and the `exitBeforeEnter` prop. Lets make an issue for this to investigate at a later date.

**Reply by bessim-dev** [reply]
@joshuaellis thx in advance and i hope i explain it well.
